ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -216,12 +217,76 @@ let assert_sr_support_operations ~__context ~vdi_map ~remote ~local_ops
          op_supported_on_dest_sr sr remote_ops sm_record remote
      )
 
+(** [get_vdi_type vdi_ref vdi_format_map] returns the vdi type found in the 
+    [vdi_format_map] mapping for a given [vdi_ref]. If no type is found None
+    is returned. *)
+let get_vdi_type ~vdi_ref ~vdi_format_map =
+  List.assoc_opt vdi_ref vdi_format_map
+
+(** [assert_vdi_format_is_supported] checks that all VDIs in [vdi_map] are included in the list of 
+    supported image format of their corresponding SM. The type of the VDI is found in [vdi_format_map].
+    - If no VDI type is specified we just returned so no error is raised.
+    - If an SM reports an empty list of supported formats, we cannot verify compatibility and no error
+      is raised. So if the format is not actually supported, the failure will be detected later when
```

Comment:
It maybe better to raise an error if the user pass a VDI type but SM doesn't have any types set. Currently it is accepted and so the creation of the VDI will fail. It is ok and the behavior is correct I think but we can also choose to just report an error during checks if no supported image formats are available and the user pass a type... Not sure what is the best approach.